src/constructors/constructWithOptions.js

```diff
@@ -10,6 +10,21 @@ export default (css: Function) => {
       throw new Error(`Cannot create styled-component for component: ${tag}`)
     }
 
+    // Creates resolver function which wraps the logic for resolving component attrs.
+    // Currently it supports both ways for defining component attrs:
+    // - plain object -> styled.input.attrs({ example: 'value' })
+    // - factory function -> styled.input.attrs(props => ({ example: props.value })
+    const createAttrsResolver = attrs => context => {
```

Comment:
Can you turn this into a pure function please? options.attrs can be passed separately. Also you can simplify the logic and prevent future bugs by not adding resolveAttrs, but by just using the attrs option. In that case you’d wrap an object in a function that just takes the context, and then only have the function-only combination logic here. That would also remove one branch